app/controllers/locations_controller.rb

```diff
@@ -1,31 +1,43 @@
 class LocationsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]  # Add this line to enforce authentication
   before_action :set_location, only: [:show, :edit, :update, :destroy]
 
   # GET /locations
   def index
     if params[:query].present?
       @locations = Location.search_by_name_address_description(params[:query])
     else
-    @locations = Location.all
-    @markers = @locations.geocoded.map do |location|
-      {
+      @locations = Location.all
+      @markers = @locations.geocoded.map do |location| {
         lat: location.latitude,
         lng: location.longitude,
-        info_window: render_to_string(partial: "popup", locals: { location: location})
+        info_window: render_to_string(partial: "popup", locals: { location: location })
       }
       end
     end
   end
 
+  def my_locations
+    if current_user.is_host
+      @my_locations = Location.where(user: current_user)
+    else
+      redirect_to root_path, status: :unauthorized
+    end
+  end
+
   # GET /locations/:id
   def show
     @booking = Booking.new
-    @pending_booking = Booking.find_by(user: current_user, location: params[:id])
+    @pending_booking = Booking.find_by(user: current_user, location: @location)
   end
 
   # GET /locations/new
   def new
-    @location = Location.new
+    if current_user.is_host
+      @location = Location.new
+    else
+      redirect_to root_path, status: :unauthorized
+    end
```

Comment:
Similar to how we are using `set_location` on line 3, this logic could be exported to a private method, and validated with a `before_action` callback :) 
something like this: 
```
def validate_host
  redirect_to root_path, status: :unauthorized unless current_user.is_host
end
```
and the callback: 
```
before_action :validate_host, only: [:new, :edit, :destroy]
```
this way, your `new`, `edit`, and `destroy` methods can simply contain the code necessary for the host user's actions 🚀  
```suggestion
      @location = Location.new
```

Reply:
Thanks for the recommendation @sofofos! I've incorporated these changes to my code.